pkg/kubelet/oom/oom_watcher_linux_test.go

```diff
@@ -26,19 +28,151 @@ import (
+		eventsRecorded = append(eventsRecorded, event)
+
+		if len(eventsRecorded) == numExpectedOomEvents {
+			break
+		}
+	case <-time.After(10 * time.Second):
+		break
+	}
+
+	return eventsRecorded
+}
+
+// TestWatcherRecordsEventsForOomEventsCorrectContainerName verifies that we
+// only record OOM events when the container name is the one for which we want
+// to record events (i.e. /).
+func TestWatcherRecordsEventsForOomEventsCorrectContainerName(t *testing.T) {
+	// By "incorrect" container name, we mean a container name for which we
+	// don't want to record an oom event.
+	numOomEventsWithIncorrectContainerName := 1
+	oomInstancesToStream := []*oomparser.OomInstance{
+		{
+			Pid:                 1000,
+			ProcessName:         "fakeProcess",
+			TimeOfDeath:         time.Now(),
+			ContainerName:       recordEventContainerName,
+			VictimContainerName: "some-container",
+		},
+		{
+			Pid:                 1000,
+			ProcessName:         "fakeProcess",
+			TimeOfDeath:         time.Now(),
+			ContainerName:       "/dont-record-oom-event",
+			VictimContainerName: "some-container",
+		},
+	}
+	numExpectedOomEvents := len(oomInstancesToStream) - numOomEventsWithIncorrectContainerName
+
+	fakeStreamer := &fakeStreamer{
+		oomInstancesToStream: oomInstancesToStream,
+	}
+
+	fakeRecorder := record.NewFakeRecorder(numExpectedOomEvents)
+	node := &v1.ObjectReference{}
+
+	oomWatcher := &realWatcher{
+		recorder:    fakeRecorder,
+		oomStreamer: fakeStreamer,
+	}
+	assert.NoError(t, oomWatcher.Start(node))
+
+	eventsRecorded := getRecordedEvents(fakeRecorder, numExpectedOomEvents)
+	assert.Equal(t, numExpectedOomEvents, len(eventsRecorded))
+}
+
+// TestWatcherRecordsEventsForOomEventsWithAdditionalInfo verifies that our the
+// emitted event has the proper pid/process data when appropriate.
+func TestWatcherRecordsEventsForOomEventsWithAdditionalInfo(t *testing.T) {
+	// The process and event info should appear in the event message.
+	eventPid := 1000
+	processName := "fakeProcess"
+
+	oomInstancesToStream := []*oomparser.OomInstance{
+		{
+			Pid:                 eventPid,
+			ProcessName:         processName,
+			TimeOfDeath:         time.Now(),
+			ContainerName:       recordEventContainerName,
+			VictimContainerName: "some-container",
+		},
+	}
+	numExpectedOomEvents := len(oomInstancesToStream)
+
+	fakeStreamer := &fakeStreamer{
+		oomInstancesToStream: oomInstancesToStream,
+	}
+
+	fakeRecorder := record.NewFakeRecorder(numExpectedOomEvents)
+	node := &v1.ObjectReference{}
 
 	oomWatcher := &realWatcher{
 		recorder:    fakeRecorder,
-		oomStreamer: oomStreamer,
+		oomStreamer: fakeStreamer,
 	}
 	assert.NoError(t, oomWatcher.Start(node))
+
+	eventsRecorded := getRecordedEvents(fakeRecorder, numExpectedOomEvents)
+
+	assert.Equal(t, numExpectedOomEvents, len(eventsRecorded))
+	assert.Contains(t, eventsRecorded[0], fmt.Sprintf("pid: %d", eventPid))
+	assert.Contains(t, eventsRecorded[0], fmt.Sprintf("victim process: %s", processName))
```

Comment:
should we ensure the event contains SystemOOM?

Reply:
Sure! Diff is updated :)